Libs/MRML/Core/vtkMRMLSegmentationDisplayNode.cxx

```diff
@@ -365,62 +368,121 @@ void vtkMRMLSegmentationDisplayNode::SetSegmentDisplayProperties(std::string seg
+  this->UpdateSegmentList();
+  SegmentDisplayPropertiesMap::iterator propsIt = this->SegmentationDisplayProperties.find(segmentID);
+  if ( propsIt != this->SegmentationDisplayProperties.end()
+    && propsIt->second.OverrideColor[0] != SEGMENT_COLOR_NO_OVERRIDE
+    && propsIt->second.OverrideColor[1] != SEGMENT_COLOR_NO_OVERRIDE
+    && propsIt->second.OverrideColor[2] != SEGMENT_COLOR_NO_OVERRIDE )
     {
-    this->Modified();
+    // If found and overridden, then return the override color
+    color[0] = propsIt->second.OverrideColor[0];
+    color[1] = propsIt->second.OverrideColor[1];
+    color[2] = propsIt->second.OverrideColor[2];
     }
 }
 
 //---------------------------------------------------------------------------
-vtkVector3d vtkMRMLSegmentationDisplayNode::GetSegmentColor(std::string segmentID)
+void vtkMRMLSegmentationDisplayNode::GetSegmentColor(std::string segmentID, double &r, double &g, double &b)
+{
+  double color[3] = {vtkSegment::SEGMENT_COLOR_INVALID[0], vtkSegment::SEGMENT_COLOR_INVALID[1], vtkSegment::SEGMENT_COLOR_INVALID[2]};
+  this->GetSegmentColor(segmentID, color);
+
+  r = color[0];
+  g = color[1];
+  b = color[2];
+}
+
+//---------------------------------------------------------------------------
+vtkVector3d vtkMRMLSegmentationDisplayNode::GetSegmentOverrideColor(std::string segmentID)
 {
   this->UpdateSegmentList();
   SegmentDisplayPropertiesMap::iterator propsIt = this->SegmentationDisplayProperties.find(segmentID);
   if (propsIt == this->SegmentationDisplayProperties.end())
     {
-    vtkErrorMacro("GetSegmentColor: No display properties found for segment with ID " << segmentID);
-    vtkVector3d color(vtkSegment::SEGMENT_COLOR_VALUE_INVALID[0], vtkSegment::SEGMENT_COLOR_VALUE_INVALID[1], vtkSegment::SEGMENT_COLOR_VALUE_INVALID[2]);
+    vtkErrorMacro("GetSegmentOverrideColor: No display properties found for segment with ID " << segmentID);
+    vtkVector3d color(SEGMENT_COLOR_NO_OVERRIDE, SEGMENT_COLOR_NO_OVERRIDE, SEGMENT_COLOR_NO_OVERRIDE);
     return color;
     }
 
-  vtkVector3d color(propsIt->second.Color[0], propsIt->second.Color[1], propsIt->second.Color[2]);
+  vtkVector3d color(propsIt->second.OverrideColor[0], propsIt->second.OverrideColor[1], propsIt->second.OverrideColor[2]);
   return color;
 }
 
 //---------------------------------------------------------------------------
-void vtkMRMLSegmentationDisplayNode::SetSegmentColor(std::string segmentID, double r, double g, double b)
+void vtkMRMLSegmentationDisplayNode::SetSegmentOverrideColor(std::string segmentID, double r, double g, double b)
 {
-  // Set color in display properties
+  // Set override color in display properties
   SegmentDisplayProperties properties;
-  this->GetSegmentDisplayProperties(segmentID, properties);
-  properties.Color[0] = r;
-  properties.Color[1] = g;
-  properties.Color[2] = b;
-  this->SetSegmentDisplayProperties(segmentID, properties);
+  if (this->GetSegmentDisplayProperties(segmentID, properties))
```

Comment:
we should allow setting segment override color no matter if a segmentation node is set or not (for example, display node may be read from XML before the segment node is read)